lib/simplifyExpression.js

```diff
@@ -67,13 +68,6 @@ function stepThrough(node, debug=false) {
     }
   }
 
-  // Update the last step, in case we returned something simplified the last
-  // time that didn't count as a step.
-  if (steps.length > 0) {
```

Comment:
This change actually breaks this test `subtraction support -------5 -> -5`, but this is a confusing step. Can you explain to me what's going on and why this is necessary?